optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/CommonUtils.java

```diff
@@ -69,69 +67,80 @@ else if (!bIter.hasNext()) {
                         if (Integer.compare(aChar, bChar) != 0) {
                             return Integer.compare(aChar, bChar);
                         }
-                    }
-                    else if (aHasChar) {
+                    } else if (aHasChar) {
                         return 1;
-                    }
-                    else if (bHasChar) {
+                    } else if (bHasChar) {
                         return -1;
-                    }
-                    else {
+                    } else {
                         return 0;
                     }
-                    
-                }
-                else {
+
+                } else {
                     return Integer.compare(aChar, bChar);
                 }
-            }
-            else {
+            } else {
                 if (Integer.compare(aChar, bChar) != 0) {
                     return Integer.compare(aChar, bChar);
                 }
             }
         }
     }
-    
+
     public static boolean isDigit(int a) {
         char[] chars = Character.toChars(a);
         return chars.length == 1 && Character.isDigit(chars[0]);
     }
-    
+
+    public static boolean isInteger(String str) {
```

Comment:
@mbarkley Don't we have any utility method on Errai, GWT or java itself to check this?

Reply:
You can use `Integer.parseInt` in GWT code. For reference, [here](http://www.gwtproject.org/doc/latest/RefJreEmulation.html) are the parts of the JRE emulated in GWT.